pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -217,66 +290,90 @@ func (a *Operator) ensureTenantRBAC(operatorNamespace, targetNamespace string, c
-	namespaces := targetNamespaces
-	if len(targetNamespaces) == 1 && targetNamespaces[0] == corev1.NamespaceAll {
+func (a *Operator) copyCsvToTargetNamespace(csv *v1alpha1.ClusterServiceVersion, operatorGroup *v1alpha2.OperatorGroup) error {
+	namespaces := make([]string, 0)
+	if len(operatorGroup.Status.Namespaces) == 1 && operatorGroup.Status.Namespaces[0] == corev1.NamespaceAll {
 		namespaceObjs, err := a.lister.CoreV1().NamespaceLister().List(labels.Everything())
 		if err != nil {
 			return err
 		}
-		namespaces = []string{}
 		for _, ns := range namespaceObjs {
 			namespaces = append(namespaces, ns.GetName())
 		}
+	} else {
+		namespaces = operatorGroup.Status.Namespaces
 	}
 
+	logger := a.Log.WithField("operator-ns", operatorGroup.GetNamespace())
 	for _, ns := range namespaces {
 		if ns == operatorGroup.GetNamespace() {
 			continue
 		}
+		logger = logger.WithField("target-ns", ns)
+
 		fetchedCSV, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(ns).Get(csv.GetName())
-		if k8serrors.IsAlreadyExists(err) {
-			log.Debugf("Found existing CSV (%v), checking annotations", fetchedCSV.GetName())
-			if reflect.DeepEqual(fetchedCSV.Annotations, csv.Annotations) == false {
+
+		logger = logger.WithField("csv", csv.GetName())
+		if fetchedCSV != nil {
+			logger.Debug("checking annotations")
+			if !reflect.DeepEqual(fetchedCSV.Annotations, csv.Annotations) {
 				fetchedCSV.Annotations = csv.Annotations
-				// CRDs don't support strategic merge patching, but in the future if they do this should be updated to patch
-				log.Debugf("Updating CSV %v in namespace %v", fetchedCSV.GetName(), ns)
+				// CRs don't support strategic merge patching, but in the future if they do this should be updated to patch
+				logger.Debug("updating target CSV")
 				if _, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(ns).Update(fetchedCSV); err != nil {
-					log.Errorf("Update CSV in target namespace failed: %v", err)
+					logger.WithError(err).Error("update target CSV failed")
 					return err
 				}
 			}
+
+			logger.Debug("checking status")
+			newCSV := fetchedCSV.DeepCopy()
+			newCSV.Status = csv.Status
+			newCSV.Status.Reason = v1alpha1.CSVReasonCopied
+			newCSV.Status.Message = fmt.Sprintf("The operator is running in %s but is managing this namespace", csv.GetNamespace())
+
+			if !reflect.DeepEqual(fetchedCSV.Status, newCSV.Status) {
+				logger.Debug("updating status")
+				newCSV.SetNamespace(ns)
+				newCSV.Status.LastUpdateTime = timeNow()
+				if _, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(ns).UpdateStatus(newCSV); err != nil {
+					logger.WithError(err).Error("status update for target CSV failed")
+					return err
+				}
+			}
+
 			continue
 		} else if k8serrors.IsNotFound(err) {
 			newCSV := csv.DeepCopy()
 			newCSV.SetNamespace(ns)
 			newCSV.SetResourceVersion("")
```

Comment:
`ObjectMeta.ResourceVersion` is read only so I don't think this needs to be cleared here.